iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/agent/task/connection/PipeEventCollector.java

```diff
@@ -144,32 +141,9 @@ private void parseAndCollectEvent(final PipeTsFileInsertionEvent sourceEvent) th
     }
 
     try {
-      final Iterable<TabletInsertionEvent> iterable = sourceEvent.toTabletInsertionEvents();
-      final Iterator<TabletInsertionEvent> iterator = iterable.iterator();
-      while (iterator.hasNext()) {
-        final TabletInsertionEvent parsedEvent = iterator.next();
-        int retryCount = 0;
-        while (true) {
-          try {
-            collectParsedRawTableEvent((PipeRawTabletInsertionEvent) parsedEvent);
-            break;
-          } catch (final PipeRuntimeOutOfMemoryCriticalException e) {
-            if (retryCount++ % 100 == 0) {
-              LOGGER.warn(
-                  "parseAndCollectEvent: failed to allocate memory for parsing TsFile {}, retry count is {}, will keep retrying.",
-                  sourceEvent.getTsFile(),
-                  retryCount,
-                  e);
-            } else if (LOGGER.isDebugEnabled()) {
-              LOGGER.debug(
-                  "parseAndCollectEvent: failed to allocate memory for parsing TsFile {}, retry count is {}, will keep retrying.",
-                  sourceEvent.getTsFile(),
-                  retryCount,
-                  e);
-            }
-          }
-        }
-      }
+      sourceEvent.consumeTabletInsertionEventsWithRetry(
+          this::collectParsedRawTableEvent,
+          this.getClass().getSimpleName() + "::parseAndCollectEvent");
```

Comment:
Better use a "static final" variable instead of compute this string each time...